core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java

```diff
@@ -133,14 +133,14 @@ public DistributedCacheStream(Address localAddress, boolean parallel, Distributi
     * Stream from an IntStream, DoubleStream etc.
     * @param other other instance of {@link AbstractCacheStream} to copy details from
     */
-   protected DistributedCacheStream(AbstractCacheStream other) {
+   public DistributedCacheStream(AbstractCacheStream other) {
```

Comment:
Hrmm it seems these changes snuck in?

Reply:
I would like to reuse the `DistributedCacheStream` implementation. See https://github.com/infinispan/infinispan/pull/6490/files#diff-b3f3c6e34c53f47dfb4deaa44a2d3c50R69

Reply:
I don't think that assertion is really needed as the setup part of the test already guarantees that. (ie. this isn't related to this test) If there was a problem with that it would be a bug with the configuration class.

Reply:
If the configuration is
```java
   @Override
   protected void enhanceConfiguration(ConfigurationBuilder builder) {
      builder.persistence().addStore(DummyInMemoryStoreConfigurationBuilder.class).async().enable();
   }
```
the test will pass.
I would like make sure that we are using `storeConfiguration.shared() && storeConfiguration.async().enabled()`

Reply:
And now I remember the main reason behind the change in the interface.
I cannot mock `DistributedCacheStream` because of 
```java
   public DistributedCacheStream(AbstractCacheStream other) {
      super(other);

      Configuration configuration = registry.getComponent(Configuration.class);
      writeBehindShared = hasWriteBehindSharedStore(configuration.persistence());
   }
```

Reply:
> If the configuration is
> 
> ```java
>    @Override
>    protected void enhanceConfiguration(ConfigurationBuilder builder) {
>       builder.persistence().addStore(DummyInMemoryStoreConfigurationBuilder.class).async().enable();
>    }
> ```
> 
> the test will pass.
> I would like make sure that we are using `storeConfiguration.shared() && storeConfiguration.async().enabled()`

This is a problem then. All entries shouldn't be returned in that case, there must be a different setup issue. I can look a bit closer.

Reply:
> And now I remember the main reason behind the change in the interface.
> I cannot mock `DistributedCacheStream` because of
> 
> ```java
>    public DistributedCacheStream(AbstractCacheStream other) {
>       super(other);
> 
>       Configuration configuration = registry.getComponent(Configuration.class);
>       writeBehindShared = hasWriteBehindSharedStore(configuration.persistence());
>    }
> ```

I don't understand why you need to mock this class.

Reply:
> I don't understand why you need to mock this class.

The test is to test a write-behind shared store, the only way that I found that we are really testing a write-behind share store is mocking `DistributedCacheStream` and doing an assert against the method `hasWriteBehindSharedStore`

> This is a problem then. All entries shouldn't be returned in that case, there must be a different setup issue. I can look a bit closer.

https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java#L659 will return `true`

Reply:
@diegolovison If you don't trust that the test is set up with a write-behind store you can use `cache.getConfiguration().persistence().usingAsyncStore()` in the test.